pkg/apisix/route.go

```diff
@@ -41,16 +41,24 @@ func newRouteClient(c *cluster) Route {
 }
 
 // Get returns the Route.
+func (r *routeClient) Get(ctx context.Context, name string) (*v1.Route, error) {
+	return r.get(ctx, id.GenID(name), name)
+}
+
+// GetByName returns the Route with route name
+func (r *routeClient) GetByID(ctx context.Context, realID string) (*v1.Route, error) {
+	return r.get(ctx, realID, realID)
+}
+
 // FIXME, currently if caller pass a non-existent resource, the Get always passes
 // through cache.
-func (r *routeClient) Get(ctx context.Context, name string) (*v1.Route, error) {
+func (r *routeClient) get(ctx context.Context, id string, name string) (*v1.Route, error) {
 	log.Debugw("try to look up route",
 		zap.String("name", name),
 		zap.String("url", r.url),
 		zap.String("cluster", "default"),
 	)
-	rid := id.GenID(name)
-	route, err := r.cluster.cache.GetRoute(rid)
+	route, err := r.cluster.cache.GetRoute(id)
```

Comment:
It still queries the cache. Does this PR still work in progress?

Reply:
Fixed. 
BTW, I noticed only consumers are not imported to cache. Is that on purpose?

Reply:
Consumers will be saved into cache.

Reply:
Got it. Will fixed it in another PR